### PR TITLE
feat: BNGP-3571 Update Legal Agreement parties flow

### PR DESCRIPTION
**Pull request by KiranVarma02**

### CODE REVIEW

**Review by pwadmore-ea — CHANGES_REQUESTED** [review]
- I cannot see any reference to the local planning authority picklist referenced from AC1 on the Jira story. The current implementation allows free text entry which does not match the acceptance criteria. I was expecting something like a new database table to hold the picklist and use of that picklist to populate HTML options similar to the prototype (https://github.com/defra-design/biodiversity-net-gain-prototype/blob/main/app/views/register-application/v21/legal-agreement-lpa-add.html)
- Changing a local planning authority appears to add a new item rather than replacing the existing item. Resolution of this issue should be accompanied by additional unit tests.
- Is packages/webapp/src/views/land/legal-party-add-start.html still required or can it be deleted in this pull request or an associated pull request for legal agreement changes?

**Comment by KiranVarma02** [comment]
> * does not match the acceptance criteria

1. Next commit will address first issue but need your thoughts/input please
2. Fixed in next commit
3. legal-party-add-start.html is still required because it has different heading and button text

File: packages/webapp/src/routes/__tests__/land/check-planning-authorities.spec.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    localPlanningAuthorities = require('../../land/check-planning-authorities.js')
+  })
+
+  describe('GET', () => {
+    it(`should render the ${url.substring(1)} view`, async () => {
+      await submitGetRequest({ url })
+    })
+
+    it('should show all landowners that are added', async () => {
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply packages/webapp/src/routes/__tests__/land/check-planning-authorities.spec.js]: done

File: packages/webapp/src/routes/__tests__/land/check-planning-authorities.spec.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+      expect(viewResult).toEqual(constants.routes.NEED_ADD_ALL_PLANNING_AUTHORITIES)
+    })
+  })
+
+  describe('POST', () => {
+    it('Should continue journey to ADD_PLANNING_AUTHORITY if yes is chosen', async () => {
+      const request = {
+        yar: redisMap,
+        payload: { addAnotherplanningAuthority: 'yes' }
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply packages/webapp/src/routes/__tests__/land/check-planning-authorities.spec.js]: done

**Comment by pwadmore-ea** [comment]
@KiranVarma02  - I have a couple of additional comments:

- Display for a single local planning authority should not include a comma and a question should end with a question mark.
image
- Accessible autocomplete integration for local planning authorities does not cover this error condition on the copydeck:

If text entered doesn’t match what’s in the list
 No match for “[text entered]” – select a local planning authority from the list

Currently, an invalid local planning authority  is accepted.

image

**Comment by pwadmore-ea** [comment]
@shaundefra  - Approving this pull request for merge to the base branch  for legal agreement updates (feat/legal-agreement-base-branch) so that associated functionality can be merged . Preventing the submission of invalid local planning authorities and duplicate local planning authorities will need addressing in subsequent pull requests

**Review by pwadmore-ea — APPROVED** [review]
Approved based on known issues to be resolved in subsequent pull requests.

**Pull request merged**